ext/openssl/openssl.c

```diff
@@ -5309,6 +5337,673 @@ PHP_FUNCTION(openssl_pkcs7_decrypt)
+			if (CMS_final(cms, infile, NULL, (unsigned int) flags) != 1) {
+				php_openssl_store_errors();
+				goto clean_exit;
+			}
+			if (i2d_CMS_bio(outfile, cms) != 1) {
+				php_openssl_store_errors();
+				goto clean_exit;
+			}
+			break;
+		case ENCODING_PEM:
+			if (CMS_final(cms, infile, NULL, (unsigned int) flags) != 1) {
+				php_openssl_store_errors();
+				goto clean_exit;
+			}
+			if (PEM_write_bio_CMS(outfile, cms) == 0) {
+				php_openssl_store_errors();
+				goto clean_exit;
+			}
+			break;
+		default:
+			php_error_docref(NULL, E_WARNING, "Unknown OPENSSL encoding");
+			goto clean_exit;
+	}
+	RETVAL_TRUE;
+
+clean_exit:
+	if (cms) {
+		CMS_ContentInfo_free(cms);
+	}
+	if (infile) {
+		BIO_free(infile);
+	}
+	if (outfile) {
+		BIO_free(outfile);
+	}
+	if (others) {
+		sk_X509_pop_free(others, X509_free);
+	}
+	EVP_PKEY_free(privkey);
+	if (cert && certresource == NULL) {
+		X509_free(cert);
+	}
+}
+/* }}} */
+
+/* {{{ proto bool openssl_cms_decrypt(string infilename, string outfilename, mixed recipcert [, mixed recipkey])
+   Decrypts the S/MIME message in the file name infilename and output the results to the file name outfilename.  recipcert is a CERT for one of the recipients. recipkey specifies the private key matching recipcert, if recipcert does not include the key */
+
+PHP_FUNCTION(openssl_cms_decrypt)
+{
+	zval * recipcert, * recipkey = NULL;
+	X509 * cert = NULL;
+	EVP_PKEY * key = NULL;
+	zend_resource *certresval, *keyresval;
+	BIO * in = NULL, * out = NULL, * datain = NULL;
+	CMS_ContentInfo * cms = NULL;
+	char * infilename;
+	size_t infilename_len;
+	char * outfilename;
+	size_t outfilename_len;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "ppz|z", &infilename, &infilename_len,
+				&outfilename, &outfilename_len, &recipcert, &recipkey) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	RETVAL_FALSE;
+
+	cert = php_openssl_x509_from_zval(recipcert, 0, &certresval);
+	if (cert == NULL) {
+		php_error_docref(NULL, E_WARNING, "Unable to coerce parameter 3 to x509 cert");
+		goto clean_exit;
+	}
+
+	key = php_openssl_evp_from_zval(recipkey ? recipkey : recipcert, 0, "", 0, 0, &keyresval);
+	if (key == NULL) {
+		if (!EG(exception)) {
+			php_error_docref(NULL, E_WARNING, "Unable to get private key");
+		}
+		goto clean_exit;
+	}
+
+	if (php_openssl_open_base_dir_chk(infilename) || php_openssl_open_base_dir_chk(outfilename)) {
+		goto clean_exit;
+	}
+
+	in = BIO_new_file(infilename, PHP_OPENSSL_BIO_MODE_R(CMS_BINARY));
+	if (in == NULL) {
+		php_openssl_store_errors();
+		goto clean_exit;
+	}
+	out = BIO_new_file(outfilename, PHP_OPENSSL_BIO_MODE_W(CMS_BINARY));
+	if (out == NULL) {
+		php_openssl_store_errors();
+		goto clean_exit;
+	}
+
+	cms = SMIME_read_CMS(in, &datain);
+
+	if (cms == NULL) {
+		php_openssl_store_errors();
+		goto clean_exit;
+	}
+	if (CMS_decrypt(cms, key, cert, NULL, out, CMS_DETACHED)) {
+		RETVAL_TRUE;
+	} else {
+		php_openssl_store_errors();
+	}
+clean_exit:
+	CMS_ContentInfo_free(cms);
```

Comment:
so does this require NULL check or not?

Reply:
by documentation you are correct.  Fixed.